### PR TITLE
Add UTF-8 validation

### PR DESCRIPTION
**Pull request by pavel-esir**
- Add UTF-8 Validation operation. Behavior is aligned with python `str.decode()` method.

- Example `b'A\xC3\x28B'` contains invalid UTF-8 characters:
If `--utf8_replace_mode="ignore"` then `b'A\xC3\x28B'` -> A)B in 
If `--utf8_replace_mode="replace"` then `b'A\xC3\x28B'` -> A�)B (invalid bytes will be replaced by �).

CVS-146122 CVS-147110

### CODE REVIEW

File: tests/layer_tests.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import pytest
+import openvino as ov
+core = ov.Core()
+
+import os, sys
+core.add_extension('/Users/pavel-esir/devel/openvino_tokenizers/build-Debug/src/libopenvino_tokenizersd.dylib')
+os.environ.update({"OV_TOKENIZER_PREBUILD_EXTENSION_PATH": "/Users/pavel-esir/devel/openvino_tokenizers/build-Debug/src/libopenvino_tokenizersd.dylib"})
+sys.path.append("/Users/pavel-esir/devel/openvino_tokenizers/python")
```
Comment (apaniukov) [inline tests/layer_tests.py:8]: `import openvino_tokenizers` or `from openvino_tokenizers import _get_factory`

Comment (pavel-esir) [reply tests/layer_tests.py:8]: ohh, sory. I forgot to delete this in the last moment

Comment (pavel-esir) [reply tests/layer_tests.py]: removed those lines

File: python/openvino_tokenizers/tokenizer_pipeline.py [hunk 2]
```diff
@@ -977,7 +978,14 @@ def get_ov_subgraph(self, input_nodes: List[Output]) -> List[Output]:
         *input_nodes, chars_node = input_nodes
         return _get_factory().create("FuzeRagged", input_nodes, {}).outputs() + [chars_node]
 
-
+@dataclass
+class UTF8ValidateStep(DecodingStep):
+    mode: UTF8ReplaceMode = UTF8ReplaceMode.IGNORE
```
Comment (apaniukov) [inline python/openvino_tokenizers/tokenizer_pipeline.py:983]: `Replace` is a better default choice.

**Review by apaniukov — APPROVED** [review]

File: python/openvino_tokenizers/tokenizer_pipeline.py (hunk 2 above)
Comment (pavel-esir) [reply python/openvino_tokenizers/tokenizer_pipeline.py]: done

**Pull request merged**